### DISCUSSION TITLE
PDM + pre-commit using pylint: Imports cannot be found

### DISCUSSION THREAD

**Discussion by lazarillo** [pdm-project/pdm#738] [Q&A]
I wrote a question in [Stack Overflow](https://stackoverflow.com/questions/69990555/python-pdm-pre-commit-using-pylint-imports-cannot-be-found), but after submitting it, I thought maybe it might be better to write it here.

In short, I am using PDM + the pre-commit tool to run black and pylint.  Black runs fine, pylint mostly runs fine.  But Pylint does not seem to be able to recognize the `__pypackages__` folder.

Because I want for this to work across the team, I cannot just supply a full path in an environment variable.  I didn't know if there were any others that might have had trouble with pylint + PDM.  I feel like this is more of a pylint issue than a PDM issue, since it is pylint that is not finding the folder.  But since this is potentially a common problem that other PDM acolytes are suffering, I thought I'd ask.

Thanks as always for the fast & excellent response!  :)

**Comment by frostming** [comment]
[Enabling PEP 582](https://pdm.fming.dev/#enable-pep-582-globally) in your shell would do the trick I guess

**Reply by lazarillo** [reply]
Thanks!  That might be pointing me to the problem:

Enabling PEP 582 -- which I have already done -- creates this:

```
> echo $PYTHONPATH
/usr/local/Cellar/pdm/1.10.0/libexec/lib/python3.10/site-packages/pdm/pep582
```
But I am using Python 3.9.

Again, I am still learning as to how PDM works.  But it seems like even though the Python version can be set by however I set up `pdm init`, PDM itself only has one version and it will be whatever is newest.  There is no other version of PDM other than `1.10.0` and no other Python library underneath it besides Python 3.10.

===================

But even if I resolved this problem by simply updating my $PYTHONPATH manually, I don't see how I can scale this across teams and repos.  I was thinking other team members could just `pdm sync`, but I don't understand how the $PYTHONPATH is then managed across team members aside from doing it manually.

**Reply by frostming** [reply]
The interesting part is the script under `/usr/local/Cellar/pdm/1.10.0/libexec/lib/python3.10/site-packages/pdm/pep582` is version-agnostic: it can work with Python >=2.7, that are what PDM supports to manage the project.

PDM manages the project by invoking the selected Python in `pdm init` in a subprocess to get environment-dependent information. So you can only install one PDM in a relative new Python while still keeping the ability to manage project using older pythons

**Reply by lazarillo** [reply]
That's cool and very elegant.

But I guess I will have to wait pylint to catch up before being able to use it as part of a pre-commit hook.  Either that or turn off the import errors...
